### PR TITLE
Adding Fluid file format README.

### PR DESCRIPTION
**Pull request by MatthiasWM**
I will have another look over this, but I believe that this is complete.

### CODE REVIEW

File: fluid/README_fd.txt [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+Every type keyword is followed by a word, which is usually interpreted as the
+C++ name of the Type, followed by an opening `{`, a list of properties, and
+a closing ‘}’. If the type has children, they are stored between another
+opening ‘{‘, followed by a list of types, followed by a closing ‘}’.
+
+The file end when there are no more types.
+
+Note: the "class" type has an additional Word following immediatley after
+      the keyword conatining the prefix for the class. So a class definition may
```
Comment (Peter0x44) [inline fluid/README_fd.txt:198]:
```suggestion
      the keyword containing the prefix for the class. So a class definition may
```

File: fluid/README_fd.txt [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+   |    +-- Fl_Value_Output
+   :
+   :
+
+
+Properties
+----------
+
+Properties are defined in a quite arbitrary manner and on a per-type basis.
+They can be inherited from super types, or be limited by 'is_some_tyep()'
```
Comment (Peter0x44) [inline fluid/README_fd.txt:281]:
```suggestion
They can be inherited from super types, or be limited by 'is_some_type()'
```

File: fluid/README_fd.txt [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+  “C” : if set, function is extern “C”
+  “return_type” <word> : C or C++ type descriptor, can start with “virtual” and/or “static”
+  ... : inherit more from Fl_Type
+
+Type codeblock <word> : C++ code, usually an 'if
+
+  "after" <word> : C++ code or comment following the closing '}'
+  ... : inherit more from Fl_Type
+
+Type "decl" <word> : C++ code to decalre a variable or cllass member
```
Comment (Peter0x44) [inline fluid/README_fd.txt:309]:
```suggestion
Type "decl" <word> : C++ code to declare a variable or class member
```

File: fluid/README_fd.txt [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+  none or "public" or "private" or "protected" : for declarations within classes
+      defaults to "prvate"
+  none or "local" or "global": for declarition in the code body
```
Comment (Peter0x44) [inline fluid/README_fd.txt:313]:
```suggestion
  none or "local" or "global": for declarations in the code body
```

File: fluid/README_fd.txt [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+Type "Fl_Widget" <word> : C++ variable name
+
+  none or "private" or "protected" : ...
+  "xywh" <word> : "{%d %d %d %d}" ...
+  "tooltip" <word> : tooltip text
+  "image" <word> : image name
+  "deimage" <word> : deactivated image name
+  "type" <word> : integer
+  "box" <word> : text or integer (see fltk boxtypes)
+  "down_box" <word> : (is_botton() or Fl_Input_choice" or is_menu_button())
```
Comment (Peter0x44) [inline fluid/README_fd.txt:350]:
```suggestion
  "down_box" <word> : (is_button() or Fl_Input_choice" or is_menu_button())
```

File: fluid/README_fd.txt [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+      except for fltk2 which is beyond the scope of thsi document.
+
+Note: fdesign files start with the text "Magic:". Fluid can read thse files,
+      but Forms/XFroms file are beyond the scope of this document.
+
+
+Options
+-------
+
+Options are usually composed of a Word, two Words, or a Word and a String. If
```
Comment (Peter0x44) [inline fluid/README_fd.txt:116]: I think "comprised" is closer to the intended meaning

File: fluid/README_fd.txt [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+
+
+All Types <word>
+
+  “label” <word> : text
+  “user_data” <word> : a value or an expression
+  “user_data_type” <word> : usually “void*” or “long”
+  “callback” <word> : a function name or a function body
+  “comment” <word> : one or many lines of text
+  “open” : group centent visible in the Fluid tree browser
```
Comment (Peter0x44) [inline fluid/README_fd.txt:292]:
```suggestion
  “open” : group content visible in the Fluid tree browser
```

File: fluid/README_fd.txt [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+  none or "private" or "protected" : ...
+  "xywh" <word> : "{%d %d %d %d}" ...
+  "tooltip" <word> : tooltip text
+  "image" <word> : image name
+  "deimage" <word> : deactivated image name
+  "type" <word> : integer
+  "box" <word> : text or integer (see fltk boxtypes)
+  "down_box" <word> : (is_botton() or Fl_Input_choice" or is_menu_button())
+      text or integer (see fltk boxtypes)
+  "value" <word> : (is_burron()) integer
```
Comment (Peter0x44) [inline fluid/README_fd.txt:352]:
```suggestion
  "value" <word> : (is_button()) integer
```

File: fluid/README_fd.txt [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+  "color" <word> :
+      If word starts with "0x", the rest of the field is a hex number.
+      If two intergers follow, this is color and selection_color (not written).
+      If one integer follows, it's the color index.
+  "selection_color" <word> : integer color index
+  "labeltype" <word> :
+      If the word is "image", TBD.
+      Or one of "NORMAL_LABEL", "SHADOW_LABEL", "ENGRAVED_LABEL",
+      "EMBOSSED_LABEL", or "NO_LABEL"
+  "labelfont" <word> : intger
```
Comment (Peter0x44) [inline fluid/README_fd.txt:363]:
```suggestion
  "labelfont" <word> : integer
```
Same typo is repeated several times below

**Review by Peter0x44 — COMMENTED** [review]
I read through and corrected all the typos I noticed

File: fluid/README_fd.txt [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+  "deimage" <word> : deactivated image name
+  "type" <word> : integer
+  "box" <word> : text or integer (see fltk boxtypes)
+  "down_box" <word> : (is_botton() or Fl_Input_choice" or is_menu_button())
+      text or integer (see fltk boxtypes)
+  "value" <word> : (is_burron()) integer
+  "value" <word> : (is_valuator(), is_spinner()) double
+  "color" <word> :
+      If word starts with "0x", the rest of the field is a hex number.
+      If two intergers follow, this is color and selection_color (not written).
```
Comment (Peter0x44) [inline fluid/README_fd.txt:356]:
```suggestion
      If two integers follow, this is color and selection_color (not written).
```
Missed this one

**Comment by Albrecht-S** [comment]
@MatthiasWM No review yet (will do that later), but docs say that the file "type" is "`.fl`" and not "`.fd`":
> FLUID edits and saves its state in **_.fl_** files.

https://www.fltk.org/doc-1.4/fluid.html#fluid_what_is_fluid

... and later in the same chapter:
> FLUID can also read **_.fd_** files produced by the `Forms` and `XForms` "fdesign" programs.

We should be consistent in the docs and use `.fl` rather than `.fd` in this document.

**Comment by MatthiasWM** [comment]
Ok, sorry you guys, you were simply too fast ;-) . I did clarify a few things and fixed many more typos. Happy to hear feedback.

**Comment by MatthiasWM** [comment]
> We should be consistent in the docs and use `.fl` rather than `.fd` in this document.

Yeah, I fixed that. Just a brain lapse. Glad I am not programming rocket ship navigations. Pluto anyone?

File: fluid/README_fl.txt [hunk 11]
```diff
@@ -0,0 +1,461 @@
...
+all elements of the format and its caveats.
+
+
+History
+-------
+
+Fluid, the Fast Light User Interface Designer was started in the 1990's loosely
+based on 'fdesign', a GUI designer that came with the 'Forms Library', later
+'XForms Library'. Fluid's .fl file format was originally compatible with the
+fdesign '.fd' format, but evolved somewhat ad'hoc to become what it is today.
```
Comment (Peter0x44) [inline fluid/README_fl.txt:15]: This is a nitpick, but I'm pretty sure "ad hoc" is correct.

**Review by Peter0x44 — COMMENTED** [review]
Looked through again for typos.

**Comment by MatthiasWM** [comment]
Thanks to you guys for sniffing out all my typos. I will commit this now to master. If there are further changes needed, please open another issue or discuss in fltk.coredev. Thanks.

Hoping to see many new translators based in .fl files.

**Pull request merged**